control/passivity.py

```diff
@@ -45,43 +84,126 @@ def ispassive(sys):
+        q = sys.noutputs
+        Q = -rho*np.eye(q, q)
+        S = 1.0/2.0*(one+rho*nu)*np.eye(q)
+        R = -nu*np.eye(m)
+        if sys.isctime():
+            off_diag = P@B - (C.T@S + C.T@Q@D)
+            return np.vstack((
+                np.hstack((A.T @ P + P@A - C.T@Q@C,  off_diag)),
+                np.hstack((off_diag.T, -(D.T@Q@D + D.T@S + S.T@D + R)))
+            ))
+        else:
+            off_diag = A.T@P@B - (C.T@S + C.T@Q@D)
+            return np.vstack((
+                np.hstack((A.T @ P  @ A - P - C.T@Q@C,  off_diag)),
+                np.hstack((off_diag.T, B.T@P@B-(D.T@Q@D + D.T@S + S.T@D + R)))
+            ))
+
+    n = sys.nstates
+
+    # LMI for passivity indices from A,B,C,D
+    sys_matrix_list = list()
+    sys_constants = -np.vstack((
+        np.hstack((np.zeros_like(A),  np.zeros_like(C.T))),
+        np.hstack((np.zeros_like(C), np.zeros_like(D))))
+    )
 
-    matrix_list = []
-    state_space_size = sys.nstates
-    for i in range(0, state_space_size):
-        for j in range(0, state_space_size):
-            if j <= i:
-                P = np.zeros_like(A)
-                P[i, j] = 1.0
-                P[j, i] = 1.0
-                matrix_list.append(make_LMI_matrix(P).flatten())
+    if rho is not None:
+        sys_matrix_list = _make_P_basis_matrices(
+            n, eps, 1.0, make_LMI_matrix)
+        sys_constants = -make_LMI_matrix(np.zeros_like(A), rho, eps, 1.0)
+    else:
+        sys_matrix_list = _make_P_basis_matrices(
+            n, 1.0, eps, make_LMI_matrix)
+        sys_constants = -make_LMI_matrix(np.zeros_like(A), eps, nu, 1.0)
 
-    coefficents = np.vstack(matrix_list).T
+    sys_coefficents = np.vstack(sys_matrix_list).T
 
-    constants = -np.vstack((
-        np.hstack((np.zeros_like(A),  - C.T)),
-        np.hstack((- C, -D - D.T)))
-    )
+    # LMI to ensure P is positive definite
+    P_matrix_list = _P_pos_def_constraint(n)
+    P_coefficents = np.vstack(P_matrix_list).T
+    P_constants = np.zeros((n, n))
 
+    # cost function
     number_of_opt_vars = int(
-        (state_space_size**2-state_space_size)/2 + state_space_size)
+        (n**2-n)/2 + n)
     c = cvx.matrix(0.0, (number_of_opt_vars, 1))
+    c = cvx.matrix(np.append(np.array(c), -1.0))
+
+    Gs = [cvx.matrix(sys_coefficents)] + [cvx.matrix(P_coefficents)]
+    hs = [cvx.matrix(sys_constants)] + [cvx.matrix(P_constants)]
 
     # crunch feasibility solution
     cvx.solvers.options['show_progress'] = False
-    sol = cvx.solvers.sdp(c,
-                          Gs=[cvx.matrix(coefficents)],
-                          hs=[cvx.matrix(constants)])
+    sol = cvx.solvers.sdp(c, Gs=Gs, hs=hs)
+    return sol["x"]
+
 
-    return (sol["x"] is not None)
+def get_output_fb_index(sys):
+    '''
+    Returns the output feedback passivity index for the input system. This is largest gain that can be placed in
+    positive feedback with a system such that the new interconnected system is passive.
+    Parameters
```

Comment:
This first line of the docstring needs to be a short phrase that describes the function, so that the table entriex in the sphinx documentation show up correctly.  See https://numpydoc.readthedocs.io/en/latest/format.html for information (in particular, "short summary" versus "extended summary").  See [`iosys.py`](https://github.com/python-control/python-control/blob/main/control/iosys.py) for lots of examples.

Also, I think you need a blank line before Parameters for Sphinx to properly parse things.